<launch>
  <arg name="robot_name" default="0" />
  <arg name="pose_x" default="0" />
  <arg name="pose_y" default="0" />
  <arg name="robot_color" default="Red" />

  <node pkg="tf" type="static_transform_publisher" name="nexus$(arg robot_name)_world"  args="$(arg pose_x) $(arg pose_y) 0.1 0 0 0 /map /nexus$(arg robot_name)/odom 100"/>
  <!-- <node pkg="tf" type="static_transform_publisher" name="nexus$(arg robot_name)_world"  args="$(arg pose_x) $(arg pose_y) 0 0 0 0 /map /nexus$(arg robot_name)/odom 100"/> -->


  <group ns="nexus$(arg robot_name)">
    <param name="tf_prefix" value="nexus$(arg robot_name)" />

    <!-- Load robot description -->
    <!-- <param command="$(find xacro)/xacro $(find marmot)/urdf/nexus_4wd_mecanum_description/urdf/nexus_4wd_mecanum.xacro robot_color:=$(arg robot_color)" name="robot_description"/> -->
    <param command="$(find xacro)/xacro $(find marmot)/urdf/simple.xacro robot_color:=$(arg robot_color)" name="robot_description"/>


    <!-- Spawn the robot -->
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
          args="-urdf -model nexus$(arg robot_name) -param robot_description -x $(arg pose_x) -y $(arg pose_y) -z 1" />


    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

    <node name="roboNode" pkg="marmot" type="graphController.py" output="screen" clear_params="true">
      <param name="name" value="$(arg robot_name)" /> 
    </node>


  </group>
  
  
  <!-- <group ns="$(arg robot_name)">
    <node name="roboNode" pkg="marmot" type="roboNode.py" output="screen" clear_params="true">
      <param name="mode" value="$(arg robot_name)" /> 
    </node>
  </group> -->

  <!-- <group ns="nexus$(arg robot_name)">
    <node name="roboController" pkg="marmot" type="roboController2.py" output="screen" clear_params="true">
      <param name="name" value="$(arg robot_name)" /> 
    </node>
    <node name="roboPID" pkg="marmot" type="roboPID2.py" output="screen" clear_params="true">
      <param name="name" value="$(arg robot_name)" /> 
    </node>
  </group> -->


</launch>
